### ISSUE TITLE
AnimationSlot bug?

### ISSUE THREAD

**Issue by TheOmnisimush** [FlaxEngine/FlaxEngine#1803]
**Setup**
1. Create AnimatedModel

2. Create AnimGraph, like this
just a simple AnimationSlot connected to Animation Output pose, without any Animation attached to it. We will call the Animation by Script
![animslot_01](https://github.com/FlaxEngine/FlaxEngine/assets/39396276/4513e500-e8a2-463d-b0a2-14a8af57ef86)

3. Create C# Code and attach to the AnimatedModel above
```
public class AnimSlotTest : Script
{
    //PUBLIC
    public Animation      animIdle      = null;      //Anim : IDLE

    //PRIVATE
    private AnimatedModel m_Model       = null;
    private int           m_sCounter    = 0;

     //::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
    /// <inheritdoc/>
    public override void OnStart()
    {
        m_Model = Actor.As<AnimatedModel>();
    }
        
    /// <inheritdoc/>
    public override void OnUpdate()
    {
        _PlayAnim();
        
    }

    //@Brief : Process Animation by script
    private void _PlayAnim()
    {
        //Error Check
        if (m_Model == null)
            return;

        //Process
        string pSlotName = "CustomSlot";
        if (!m_Model.IsPlayingSlotAnimation(pSlotName, animIdle))
        {
            //Play Animation with LoopCount = -1 (infinite loop)
           m_Model.PlaySlotAnimation(pSlotName, animIdle, 1.0f, 0.2f, 0.2f, -1);

           //Show Log in Console 
           Debug.Log($"AnimSlot : {pSlotName}, {m_sCounter}\nPlay anim {animIdle.Path}");

           //Iterate Counter to see howmany times we call "PlaySlotAnimation()"
           m_sCounter ++;
        }
    }

}
```
Basically the code will play “animIdle” in a forever loop (loopCount=-1) Animation at AnimationSlot=“CustomSlot”, if the Slot is not playing animation.

4. Test Result
![anim_slot_](https://github.com/FlaxEngine/FlaxEngine/assets/39396276/6d86d4be-5f07-4c54-8db5-6266b16ccb0a)
![animslot_00](https://github.com/FlaxEngine/FlaxEngine/assets/39396276/799a2c6c-a3b5-4711-8676-8bda0fd6cff1)

The “PlaySlotAnimation()” is called 3 times :
1st time (m_sCounter=0) → no animation played
2nd time (m_sCounter=1) → animation played, but somehow not looped although we set loopCount=-1
3rd time (m_sCounter=2) → animation played correctly with loopCount=-1

**Expected Result:**
I expected the code to just call “PlayAnimationSlot()” once, but it is called 3 times,
I kind of understand if the first time the Animation is not registered yet, so IsPlayingAnimationSlot() give a false result, but the 2nd time is weird because it does not play the correct parameter I set in the function.
Is this a bug?

**Flax version:**
1.6.6344

**Comment by mafiesto4** [comment]
For me it prints:
```
[ 00:01:45.207 ]: [Info] [PlayMode] Enter
[ 00:01:45.208 ]: [Info] AnimSlot : CustomSlot, 0
Play anim /Users/wojtek/Flax/FlaxSamples/ParticlesFeaturesTour/Content/Models/Examples/anims/Rifle Idle.flax
[ 00:01:45.267 ]: [Info] AnimSlot : CustomSlot, 1
Play anim /Users/wojtek/Flax/FlaxSamples/ParticlesFeaturesTour/Content/Models/Examples/anims/Rifle Idle.flax
[ 00:01:49.290 ]: [Info] [PlayMode] Stop
```

So it looks different, maybe it's caused by the difference in game update rate or chance of Animated Model to tick too late in your case or most likely some asset loading event that caused graph instance to reset.

**Comment by mafiesto4** [comment]
Actually, the proper flow would be to play animation only twice so `IsPlayingSlotAnimation` should return `true` on 2nd and sequential calls. I fixed this in 0b7c18763006a7442a84c03ba43b60c96d3b9408. Now it will work just fine.
